From 95bebc0877be09da2c2488e06ec5b455d5ff7fe7 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Mon, 21 Jul 2014 18:21:13 +0800
Subject: [PATCH] mmc: rk_sdmmc: add dt-bindings in rk312x.dtsi (1)Default
 disabled (2)No pinctrl and no board-relative dts perameter support

---
 arch/arm/boot/dts/rk312x.dtsi | 56 +++++++++++++++++++++++++++++++++++
 1 file changed, 56 insertions(+)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 5cc92dae68e5..3737686a6ec9 100755
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -221,6 +221,62 @@
 		rockchip,check-idle = <1>;
 		status = "disabled";
 	};
+	emmc: rksdmmc@1021c000 {
+		compatible = "rockchip,rk_mmc", "rockchip,rk312x-sdmmc";
+                reg = <0x1021c000 0x4000>;
+                interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
+                #address-cells = <1>;
+                #size-cells = <0>;
+                //pinctrl-names = "default",,"suspend";
+                //pinctrl-0 = <&sd0_clk &sd0_cmd &sd0_cd &sd0_wp &sd0_pwr &sd0_bus1 &sd0_bus4>;
+                clocks = <&clk_emmc>, <&clk_gates7 0>;
+                clock-names = "clk_mmc", "hclk_mmc";
+                dmas = <&pdma 12>;
+                dma-names = "dw_mci";
+                num-slots = <1>;
+                fifo-depth = <0x100>;
+                bus-width = <8>;
+		status = "disabled";
+        };
+
+
+        sdmmc: rksdmmc@10214000 {
+                compatible = "rockchip,rk_mmc", "rockchip,rk3036-sdmmc";
+                reg = <0x10214000 0x4000>;
+                interrupts = <GIC_SPI 14 IRQ_TYPE_LEVEL_HIGH>;
+                #address-cells = <1>;
+                #size-cells = <0>;
+                //pinctrl-names = "default", "idle";
+                //pinctrl-0 = <&sdmmc0_clk &sdmmc0_cmd &sdmmc0_dectn &sdmmc0_bus4>;
+                //pinctrl-1 = <&sdmmc0_gpio>;
+                //cd-gpios = <&gpio1 GPIO_C1 GPIO_ACTIVE_HIGH>;/*CD GPIO*/
+                clocks = <&clk_sdmmc0>, <&clk_gates5 10>;
+                clock-names = "clk_mmc", "hclk_mmc";
+                dmas = <&pdma 10>;
+                dma-names = "dw_mci";
+                num-slots = <1>;
+                fifo-depth = <0x100>;
+                bus-width = <4>;
+		status = "disabled";
+        };
+	sdio: rksdmmc@10218000 {
+                compatible = "rockchip,rk_mmc", "rockchip,rk3036-sdmmc";
+                reg = <0x10218000 0x4000>;
+                interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+                #address-cells = <1>;
+                #size-cells = <0>;
+                //pinctrl-names = "default","idle";
+                //pinctrl-0 = <&sdio0_clk &sdio0_cmd  &sdio0_bus4>;
+                //pinctrl-1 = <&sdio0_gpio>;
+                clocks = <&clk_sdio>, <&clk_gates5 11>;
+                clock-names = "clk_mmc", "hclk_mmc";
+                dmas = <&pdma 11>;
+                dma-names = "dw_mci";
+                num-slots = <1>;
+                fifo-depth = <0x100>;
+                bus-width = <4>;
+		status = "disabled";
+        };
 
 	adc: adc@2006c000 {
 		compatible = "rockchip,saradc";
-- 
2.35.3

